backend/app/core/exceptions.py

```
@@ -0,0 +1,61 @@
+"""
+Custom exception classes for SEPE Templates Comparator.
+
+This module defines the exception hierarchy used throughout the application
+for consistent error handling and reporting.
+"""
+
+
+class SEPEComparatorError(Exception):
+    """Base exception for SEPE Comparator application."""
+    
+    def __init__(self, message: str, details: str = None):
```

Ruff (0.14.0)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)